packages/hardhat-core/src/internal/hardhat-network/provider/utils/convertToRethnet.ts

```diff
@@ -164,6 +45,46 @@ export function ethereumjsBlockHeaderToRethnet(
   };
 }
 
+export function ethereumsjsHardforkToRethnet(hardfork: HardforkName): SpecId {
+  switch (hardfork) {
+    case HardforkName.FRONTIER:
+      return SpecId.Frontier;
+    case HardforkName.HOMESTEAD:
+      return SpecId.Homestead;
+    case HardforkName.DAO:
+      return SpecId.DaoFork;
+    case HardforkName.TANGERINE_WHISTLE:
+      return SpecId.Tangerine;
+    case HardforkName.SPURIOUS_DRAGON:
+      return SpecId.SpuriousDragon;
+    case HardforkName.BYZANTIUM:
+      return SpecId.Byzantium;
+    case HardforkName.CONSTANTINOPLE:
+      return SpecId.Constantinople;
+    case HardforkName.PETERSBURG:
+      return SpecId.Petersburg;
+    case HardforkName.ISTANBUL:
+      return SpecId.Istanbul;
+    case HardforkName.MUIR_GLACIER:
+      return SpecId.MuirGlacier;
+    case HardforkName.BERLIN:
+      return SpecId.Berlin;
+    case HardforkName.LONDON:
+      return SpecId.London;
+    case HardforkName.ARROW_GLACIER:
+      return SpecId.ArrowGlacier;
+    case HardforkName.GRAY_GLACIER:
+      return SpecId.GrayGlacier;
+    case HardforkName.MERGE:
+      return SpecId.Merge;
+    default:
+     const _exhaustiveCheck: never = hardfork;
```

Comment:
I added this so that we get a compilation error if we add anew `HardforkName` but forget to update this switch.